tests/ui/manual_arithmetic_check.rs

```diff
@@ -0,0 +1,16 @@
+#![warn(clippy::manual_arithmetic_check)]
+
+fn main() {
+    let a = 12u32;
+    let b = 13u32;
+
+    let result = if a > b { a - b } else { 0 };
+    //~^ ERROR: manual arithmetic check found
+    let result = if b < a { a - b } else { 0 };
+    //~^ ERROR: manual arithmetic check found
+
+    let result = if a < b { 0 } else { a - b };
+    //~^ ERROR: manual arithmetic check found
+    let result = if b > a { 0 } else { a - b };
+    //~^ ERROR: manual arithmetic check found
```

Comment:
Would be good to have a test where the block has other statements

Reply:
Done as well.

Reply:
huh, were they added? I don't see one in the test file. To clarify I mean something like `if b > a { 0 } else { println!("do something else"); a - b }` where `saturating_sub` removes side effects/changes semantics

I could be wrong but I think that this will currently lint on it because `Expr::peel_blocks` is used, which doesnt make sure that the block has no statements in its loop. `clippy_utils::peel_blocks` OTOH does do that